### PR TITLE
Ticket3267 add ioc rkndio

### PR DESCRIPTION
**Pull request by ghost**
## Related to

- [#3267](https://github.com/ISISComputingGroup/IBEX/issues/3267)
- [#3304](https://github.com/ISISComputingGroup/IBEX/issues/3304)

## Description of what has been done
- Added database files using substitutions and templates.
- Added protocol file to communicate with the device.

### CODE REVIEW

File: rkndioSup/rkndio.proto [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    out "STATUS";	
+    in "%39c";
+}
+
+get_error {
+    out "ERR";	
+    in "%39c";	
+}
+
+mismatch_get_error {
```
Comment (Tom-Willemsen) [inline rkndioSup/rkndio.proto]: Might be helpful to put a note on this record explaining why it starts with an `in` command. As someone reading through this protocol for the first time it is a bit surprising, it makes sense once you know that it is connected to an I/O interrupt record but a quick comment would make it even more obvious

File: rkndioSup/rkndio_pin_inputs.template [hunk 2]
```diff
@@ -0,0 +1,16 @@
+record (bi, "$(P)PIN_$(PIN_NUMBER)") {
```
Comment (Tom-Willemsen) [inline rkndioSup/rkndio_pin_inputs.template]: I think `PIN:1` would be more in line with our conventions (nitpick)

File: rkndioSup/rkndio_pin_inputs.template [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	field(DESC, "Digital Input Pin $(PIN_NUMBER).")
+	field(SCAN, ".1 second")
```
Comment (Tom-Willemsen) [inline rkndioSup/rkndio_pin_inputs.template]: Typically most things scan at 1 second frequency. Is there a need for such a fast scan?

Comment (ghost) [reply rkndioSup/rkndio_pin_inputs.template]: Kathryn asked for the IOC to scan at this rate.

**Comment by ghost** [comment]
`SCAN` rates on PVs are specified by Kathryn.

**Pull request merged**